torchvision/models/detection/_utils.py

```diff
@@ -201,16 +202,14 @@ def decode_single(self, rel_codes, boxes):
         pred_w = torch.exp(dw) * widths[:, None]
         pred_h = torch.exp(dh) * heights[:, None]
 
-        pred_boxes = torch.zeros_like(rel_codes)
-        # x1
-        pred_boxes[:, 0::4] = pred_ctr_x - 0.5 * pred_w
-        # y1
-        pred_boxes[:, 1::4] = pred_ctr_y - 0.5 * pred_h
-        # x2
-        pred_boxes[:, 2::4] = pred_ctr_x + 0.5 * pred_w
-        # y2
-        pred_boxes[:, 3::4] = pred_ctr_y + 0.5 * pred_h
-
+        pred_boxes1 = pred_ctr_x - torch.tensor(0.5, dtype=pred_ctr_x.dtype) * pred_w
+        pred_boxes2 = pred_ctr_y - torch.tensor(0.5, dtype=pred_ctr_y.dtype) * pred_h
+        pred_boxes3 = pred_ctr_x + torch.tensor(0.5, dtype=pred_ctr_x.dtype) * pred_w
+        pred_boxes4 = pred_ctr_y + torch.tensor(0.5, dtype=pred_ctr_y.dtype) * pred_h
+        pred_boxes = torch.cat((pred_boxes1.unsqueeze(2),
```

Comment:
Thanks for the fix!

This is equivalent to `torch.stack(..., dim=2)`, and I think it's a bit more readable, would you mind making this change?

Also, I checked the performance slowdown, and it's fine for now, once `torch.cat` gets optimized for this particular corner case, there will be no perf difference.

Also, don't we need to do a `torch.flatten(1)` in here so that the behavior is the same, as `pred_boxes` was a 2d tensor, and now it's a 3d tensor